Add documentation to use  JS UDFs in private datasets (#218)
GoogleCloudPlatform/bigquery-utils: udfs/README.md

`CREATE OR REPLACE FUNCTION project_id.your_dataset.function_name() . . .`

### Using JavaScript UDFs

When creating [JavaScript UDFs](https://cloud.google.com/bigquery/docs/reference/standard-sql/user-defined-functions#javascript-udf-structure) in your dataset, you need both to create 

> the UDF and upload the javascript library to a publicly available Google Storage Bucket. 

Reviewer: ```suggestion
the UDF and optionally upload the javascript library to your own Google Storage Bucket. 
```

Reviewer: Can you please add another sentence which points out that all JavaScript UDF libraries are currently hosted publicly on gs://bqutil-lib/bq_js_libs/

Author: Done

In the following example, we show how to create the Levenshtein UDF function, that uses a JS
library, in your dataset.

1. We [obtain](https://www.npmjs.com/package/js-levenshtein) the latest version of the
`js-levenshtein` function.
2. We copy the library to our bucket:

> `gsutil cp js-levenshtein-v1.1.6.js gs://your-bucket`

Reviewer: Let's make this an easy 1 step by having them copy the JavaScript webpack-generated bundle from the publicly hosted GCS bigquery-utils bucket: gs://bqutil-lib/bq_js_libs/

`gsutil cp gs://bqutil-lib/bq_js_libs/js-levenshtein-v1.1.6.js gs://your-bucket`

Author: Done

3. We give permissions to the library:

> `gsutil acl ch -u AllUsers:R gs://your-bucket/js-levenshtein-v1.1.6.js`

Reviewer: Use [uniform bucket-level access](https://cloud.google.com/storage/docs/using-uniform-bucket-level-access) instead of ACLs

Narrow the member binding. Instead of AllUsers, put a placeholder for a group or Google identity of the users's choice and bind to the Cloud Storage Object Viewer role at the bucket or project level.

Author: Done. I've also added how to enable uniform bucket-level access.
